docs/howto/tracing.md

```diff
@@ -138,6 +174,101 @@ Note that the `charm_tracing` charm library auto-instruments all public function
 
+<!--
+## Trace a charm library
+
+> Hints:
+OTEL API usage.
+A (dummy) charm to test the charm lib.
+
+- At the top of your charm library, `import opentelemetry.trace`.
+- After the imports in your charm library, create the tracer object as
+  `tracer = opentelemetry.trace.get_tracer(name)` where the name could be your
+  charm library name, or Python module `__name__`.
+- See the [Custom spans and events](custom-spans-and-events) section above to
+  create OpenTelemetry spans and events in the key places in your charm library.
+-->
+
+<!--
+## Include tracing in a new charm
+-->
+
+<!--
+## View the trace data
+
+> Hints:
+How to install COS (link), relate your app to COS, where to point browser,
+screenshot of a trace in Grafana, some pointer about what's on the screenshot
+and how manual instrumentation (above) connects to that.
+-->
+
+## Test the feature
+
+A note that ops and charm libraries are already covered, maybe there's
+nothing left for the charm to do.
+
+Relation setup is already validated by the fact that your charm doesn't crash in other tests.
+
+If you've added custom instrumentation, it is because something important is recorded.
+It's easier to cover this with unit tests.
+
+### Write unit tests
+
+The {py:attr}`trace_data <ops.testing.Context.trace_data>` attribute
+of the {py:class}`ops.testing.Context` class of the testing framework 
+contains the list of finished spans.
+
+The following example demonstrates how to get the root span, created by Ops itself.
+
+```py
+ctx = Context(YourCharm)
+ctx.run(ctx.on.start(), State())
+main_span = next(s for s in ctx.trace_data if s.name == 'ops.main')
+```
+
+The spans are OpenTelemetry objects in memory, allowing more focused examination, here a check that span A is a parent of span B.
+
+```py
+span_a = ...
+span_b = ...
+assert span_a.context is span_b.parent
+```
+
+Or that span A is an ancestor of span C, which allows you to validate that an important logical thing C was done during some well-known process A: an Ops event representing the Juju event or a custom event, or some manually instrumented function.
+
+```py
+spans_by_id = {s.context.span_id: s for s in ctx.trace_data}
+
+def ancestors(span: ReadableSpan) -> Generator[ReadableSpan]:
+    while span.parent:
+        span = spans_by_id[span.parent.span_id]
+        yield span
+
+assert span_a in list(ancestors(span_c))
+```
+
+You can disambiguate spans using their {py:class}`instrumentation_scope <opentelemetry.sdk.util.instrumentation.InstrumentationScope>` property.
+
+```py
+# Spans from Ops
+ops_span.instrumentation_scope.name == "ops"
+ops_span.name == ...
+
+# tracer = opentelemetry.trace.get_tracer("my-charm")
+my_span.instrumentation_scope.name == "my-charm"
+my_span.name == ...
+```
+
+<!--
+### Write integration tests
+
+> Hints:
+Against the tracing-integration-tester charm (example)
+Against Tempo
+Links to COS documentation
+> Against grafana-agent -- don't use that any more
+-->
```

Comment:
If this is to be done in a later PR, then I'd rather we tracked that in a ticket than in a HTML comment.